mempool/mempool.go

```diff
@@ -1,26 +1,34 @@
 package mempool
 
 import (
-	"bytes"
-	"container/list"
-	"crypto/sha256"
 	"fmt"
-	"sync"
-	"sync/atomic"
-	"time"
-
-	"github.com/pkg/errors"
 
 	abci "github.com/tendermint/tendermint/abci/types"
-	cfg "github.com/tendermint/tendermint/config"
-	auto "github.com/tendermint/tendermint/libs/autofile"
-	"github.com/tendermint/tendermint/libs/clist"
-	cmn "github.com/tendermint/tendermint/libs/common"
-	"github.com/tendermint/tendermint/libs/log"
-	"github.com/tendermint/tendermint/proxy"
 	"github.com/tendermint/tendermint/types"
 )
 
+// Mempool defines the mempool interface.
+// Updates to the mempool need to be synchronized with committing a block so
+// apps can reset their transient state on Commit.
+type Mempool interface {
+	Lock()
```

Comment:
++ glad to see this is still here - we need to be able to claim this lock to avoid issues around recheck/commit.

See further: https://github.com/hyperledger/burrow/blob/develop/consensus/tendermint/abci/app.go#L321-L344